### ISSUE TITLE
How to know what is my KMI?

### ISSUE THREAD

**Issue by TheHippoMan** [tiann/KernelSU#983]
I want to try KernelSU on my Pixel 7 Pro running crDroid 9.8 (Android 13). I installed the KernelSU manager and when I run it, it says "Not installed. Click to install", so I assume that my device is indeed compatible with KernelSU. However, I don't know which KernelSU img file to install.

On the "Android version" page in my device's Settings, it says that my kernel version is "5.10.157-ga88edbe37d3a" and I assume that this isn't actually the KMI.  And there are several 5.10.157-based boot images on the KernelSU Assets page, and I don't know how to figure out which one to try with my device.

Given my kernel version, how can I  choose the proper KernelSU version to install on my device?

Thank you in advance.

**Comment by tiann** [comment]
Check the docs first.

**Comment by TheHippoMan** [comment]
I read the docs, and they say that the kernel name must match this pattern:
**Version.PatchLevel.SubLevel-AndroidRelease-KmiGeneration-suffix**

However, as I wroteabove, this is my kernel name: **5.10.157-ga88edbe37d3a**.

As you can see, it contains no "AndroidRelease" field, and so it doesn't match the stated kernel name pattern.

I therefore don't know which KernelSU image I should chose from the list in the Assets page, given that there are several with "5.10.157" in their name.

**Comment by tiann** [comment]
This means it is not a GKI kernel.

**Comment by TheHippoMan** [comment]
> This means it is not a GKI kernel.

Oh, I see. Thank you for explaining!

**Comment by TheHippoMan** [comment]
However ...

When I run the KernelSU app on that device, it says, "Not installed, click to install".

I thought this means that the KernelSU app does indeed think that I'm already running with an eligible GKI kernel.

That's why I came here in the first place to try to get a version of KernelSU that works on my device.

So even though my device's kernel is not named with AndroidRelease as part of its name, could it be that it's still a valid (but possibly just misnamed) GKI kernel?

**Comment by tiann** [comment]
The manager only use a regex to match the kernel name, if the name is incorrect, ask the kernel maintainer, i don't know.

**Comment by TheHippoMan** [comment]
Yes, I'll ask the kernel developer.

But please think about the following ...

The documentation here says that a valid GKI kernel name must look like this:

_Version.PatchLevel.SubLevel-**AndroidRelease**-KmiGeneration-suffix_

However, the KernelSU Manager's regex matched the following name as being a valid kernel name:

 _5.10.157-ga88edbe37d3a_

And as you can see, this name does not contain any AndroidRelease field.

In other words, the regex match is not consistent with the documentation here.

Does this mean that the regex is incorrect?
Or does this mean that the documentation is incorrect?
